tests/dummy/app/routes/onerror-with-multiple-error-handlers.js

```diff
@@ -0,0 +1,25 @@
+import Ember from "ember";
+import PromiseMixin from "ember-promise/mixins/promise";
+import ErrorHandler from "../utilities/error-handler";
+import SuccessHandler from "../utilities/success-handler";
+
+var OnErrorRoute = Ember.Route.extend({
+    model: function() {
+        var people = Ember.A();
+        PromiseMixin.xhr("/api/bad-with-handler", "GET").then(function(response) {
+            response.forEach(function(person) {
+                people.pushObject(Ember.Object.create(person));
+            });
+        }, function(response) {
+            ErrorHandler.handled = 1;
+            response.forEach(function(){});
```

Comment:
I did consider that and even had that code but for some reason went with
this. I think the `throw new Error` probably makes more sense.

On Tue, May 12, 2015 at 9:55 PM, Toran Billups <notifications@github.com>
wrote:

> In tests/dummy/app/routes/onerror-with-multiple-error-handlers.js
> <https://github.com/toranb/ember-promise/pull/3#discussion_r30199317>:
>
> > @@ -0,0 +1,25 @@
> > +import Ember from "ember";
> > +import PromiseMixin from "ember-promise/mixins/promise";
> > +import ErrorHandler from "../utilities/error-handler";
> > +import SuccessHandler from "../utilities/success-handler";
> > +
> > +var OnErrorRoute = Ember.Route.extend({
> > +    model: function() {
> > +        var people = Ember.A();
> > +        PromiseMixin.xhr("/api/bad-with-handler", "GET").then(function(response) {
> > +            response.forEach(function(person) {
> > +                people.pushObject(Ember.Object.create(person));
> > +            });
> > +        }, function(response) {
> > +            ErrorHandler.handled = 1;
> > +            response.forEach(function(){});
>
> Ah that makes sense - did you consider just doing a "threw new Error"
> instead to be more clear? Would you think this improves the readability for
> future developers reading the test code?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/toranb/ember-promise/pull/3/files#r30199317>.
>